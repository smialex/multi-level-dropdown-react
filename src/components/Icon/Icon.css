.icon{
  --icon-size: calc(60px * 0.7);
  width: var(--icon-size);
  height: var(--icon-size);
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--bg-accent);
  border-radius: 50%;
}

.icon > svg {
  width: 20px;
  height: 20px;
  fill: var(--text-color);
}

.icon.icon_transparent {
  background-color: transparent;
}