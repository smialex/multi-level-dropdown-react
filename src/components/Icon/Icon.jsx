import React from 'react';
import './Icon.css';

export const Icon = ({icon, transparent}) => {

  const styles = `icon ${transparent ? 'icon_transparent' : ''}`;

  return (
    <div className={styles}>{icon}</div>
  );
}

