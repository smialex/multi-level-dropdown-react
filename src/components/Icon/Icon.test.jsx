import React  from 'react';
import renderer from "react-test-renderer";
import { shallow, mount } from 'enzyme';
import { Icon } from "./Icon";

describe("Icon component", () => {
  test("render correctly Icon component", () => {
    const icon = renderer.create(<Icon/>).toJSON();
    expect(icon).toMatchSnapshot();
  });

  test('renders with transparent proprty', () => {
    const icon = shallow(<Icon transparent={true} />);
    expect(icon.contains(<div className="icon icon_transparent" ></div>)).toBeTruthy();
    expect(icon.find('.icon_transparent').length).toBe(1);
  });
  
  test('renders with icon proprty', () => {
    const icon = shallow(<Icon icon='ðŸŽ‚' />);
    expect(icon.text()).toBe('ðŸŽ‚');
  });

  test('check correctly icon prop', () => {
    const icon = mount(<Icon icon='ðŸŽ‚' />);
    expect(icon.prop('icon')).toBe('ðŸŽ‚');
  });
      
  
});
