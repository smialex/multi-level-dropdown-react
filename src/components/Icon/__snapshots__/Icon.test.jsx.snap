// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component render correctly Icon component 1`] = `
<div
  className="icon "
/>
`;

exports[`Icon render correctly Icon component 1`] = `
<div
  className="icon "
/>
`;
